agreement_legal/views/agreement_type.xml

```diff
@@ -5,33 +5,28 @@
     <record model="ir.ui.view" id="partner_agreement_type_list_view">
         <field name="name">Agreement Type List</field>
         <field name="model">agreement.type</field>
+        <field name="inherit_id" ref="agreement.agreement_type_list_view"/>
         <field name="arch" type="xml">
-            <tree string="Agreement Types" default_order='name'>
-                <field name="name" string="Type Name"/>
+            <field name="name" position="after">
                 <field name="agreement_subtypes_ids" string="Sub-Types"/>
-            </tree>
-        </field>
+            </field>
+       </field>
     </record>
 
     <!-- Agreement Type Form View -->
     <record model="ir.ui.view" id="partner_agreement_type_form_view">
         <field name="name">Agreement Type Form</field>
         <field name="model">agreement.type</field>
+        <field name="inherit_id" ref="agreement.agreement_type_form_view"/>
         <field name="arch" type="xml">
-            <form string="Agreement Type">
-                <sheet>
-                    <div class="oe_title">
-                        <label for="name" class="oe_edit_only"/>
-                        <h1><field name="name"/></h1>
-                    </div>
-                    <field name="agreement_subtypes_ids"
-                           nolabel="1">
-                        <tree editable="bottom">
-                            <field name="name"/>
-                        </tree>
-                    </field>
-                </sheet>
-            </form>
+            <xpath expr="//sheet" position="inside">
+                <field name="agreement_subtypes_ids"
+                       nolabel="1">
+                    <tree editable="bottom">
+                        <field name="name"/>
+                    </tree>
+                </field>
+            </xpath>
```

Comment:
@gurneyalex Partially off-topic of your PR but there is a missing part in agreement_sale as documented in the readme "This module adds Agreement field to Sales Orders"
Please could you move this view
https://github.com/OCA/contract/blob/12.0/agreement_legal_sale/views/sale_order.xml
to agreement_sale
But please add a group to agreement_template_id field please.
Thanks